Fetch result with ListPath.any().{field} cause NullPointerException

I have a model defined like this:

```
public class User {
	public List<Address> addresses;
}
```

And then, fetch with this command
`CollQueryFactory.from(QUser.user, users).where(predicate).fetchResults().getResults()` which `predicate` is defined as `QUser.user.addresses.any().city.eq("SomeCity")`

Also, `users` variable contains list of user and there is one user that has value of `addresses` field as `null` (it doesn't be an empty list).

It throws exception like this
```
java.lang.IllegalArgumentException: java.lang.reflect.InvocationTargetException

	at com.mysema.codegen.MethodEvaluator.evaluate(MethodEvaluator.java:52)
	at com.querydsl.collections.DefaultQueryEngine.evaluateMultipleSources(DefaultQueryEngine.java:116)
	at com.querydsl.collections.DefaultQueryEngine.count(DefaultQueryEngine.java:57)
	at com.querydsl.collections.AbstractCollQuery.fetchResults(AbstractCollQuery.java:189)
	...
Caused by: java.lang.reflect.InvocationTargetException
	at com.mysema.codegen.MethodEvaluator.evaluate(MethodEvaluator.java:48)
	... 32 more
Caused by: java.lang.NullPointerException
	at com.querydsl.collections.CollQueryFunctions.leftJoin(CollQueryFunctions.java:198)
	at Q_715591299_65821278_1275614662_1195259493.eval(Unknown Source)
	... 37 more
```

PS. I'm using querydsl 4.1.2

--- a/querydsl-collections/src/main/java/com/querydsl/collections/CollQueryFunctions.java
+++ b/querydsl-collections/src/main/java/com/querydsl/collections/CollQueryFunctions.java
@@ -195,7 +195,7 @@ public final class CollQueryFunctions {
     }
 
     public static <T> Collection<T> leftJoin(Collection<T> coll) {
-        if (coll.isEmpty()) {
+        if (coll == null || coll.isEmpty()) {
             @SuppressWarnings("unchecked") // List only contains null
             Collection<T> rv = (Collection<T>) nullList;
             return rv;

--- a/querydsl-collections/src/test/java/com/querydsl/collections/CollectionAnyTest.java
+++ b/querydsl-collections/src/test/java/com/querydsl/collections/CollectionAnyTest.java
@@ -10,6 +10,15 @@ import org.junit.Test;
 public class CollectionAnyTest extends AbstractQueryTest {
 
     @Test
+    public void any_null() {
+        Cat a = new Cat("a");
+        a.setKittens(null);
+
+        assertEquals(0, CollQueryFactory.from(cat, Arrays.asList(a))
+                .where(cat.kittens.any().name.startsWith("a")).fetchCount());
+    }
+
+    @Test
     public void any_in_projection() {
         Cat a = new Cat("a");
         Cat aa = new Cat("aa");

